sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -112,7 +112,12 @@ class SimpleFunctionRegistry extends FunctionRegistry {
         throw new AnalysisException(s"undefined function $name")
       }
     }
-    func(children)
+    try {
+      func(children)
+    } catch {
+      case _: ClassCastException =>
```

Comment:
Sorry, I made mistake. I realized the original exception happen in the stack trace that does not execute the body.

```
Caused by: java.lang.ClassCastException: org.apache.spark.sql.UDFSuite$$anonfun$9$$anonfun$29$$anonfun$apply$1 cannot be cast to scala.Function3
	at org.apache.spark.sql.catalyst.expressions.ScalaUDF.<init>(ScalaUDF.scala:109)
	at org.apache.spark.sql.UDFRegistration.org$apache$spark$sql$UDFRegistration$$builder$4(UDFRegistration.scala:160)
	at org.apache.spark.sql.UDFRegistration$$anonfun$register$4.apply(UDFRegistration.scala:161)
	at org.apache.spark.sql.UDFRegistration$$anonfun$register$4.apply(UDFRegistration.scala:161)
	at org.apache.spark.sql.catalyst.analysis.SimpleFunctionRegistry.lookupFunction(FunctionRegistry.scala:115)
...
```